src/csharp/Microsoft.Spark/Interop/Ipc/JsonSerDe.cs

```diff
@@ -2,69 +2,85 @@
             {
-                if (property.Value is JObject elem)
+                using (var writer = new Utf8JsonWriter(stream))
                 {
-                    sortedJObject.Add(property.Name, elem.SortProperties());
-                }
-                else if (property.Value is JArray arrayElem)
-                {
-                    sortedJObject.Add(property.Name, arrayElem.SortProperties());
-                }
-                else
-                {
-                    sortedJObject.Add(property.Name, property.Value);
+                    writer.WriteStartObject();
+                    foreach (JsonProperty prop in jObject.EnumerateObject().OrderBy(p => p.Name))
+                    {
+                        writer.WritePropertyName(prop.Name);
+                        if (prop.Value.ValueKind == JsonValueKind.Object)
+                        {
+                            prop.Value.SortProperties().WriteTo(writer);
+                        }
+                        else if (prop.Value.ValueKind == JsonValueKind.Array)
+                        {
+                            prop.Value.SortArrayProperties().WriteTo(writer);
+                        }
+                        else
+                        {
+                            prop.Value.WriteTo(writer);
+                        }
+                    }
+                    writer.WriteEndObject();
+                    writer.Flush();               
+                    return JsonDocument.Parse(stream.ToArray()).RootElement;
                 }
             }
-
-            return sortedJObject;
         }
 
         /// <summary>
         /// Extend method to sort items in a JSON array by keys.
         /// </summary>
-        public static JArray SortProperties(this JArray jArray)
+        public static JsonElement SortArrayProperties(this JsonElement jArray)
         {
-            if (jArray.Count == 0)
+            if (jArray.GetArrayLength() == 0)
             {
                 return jArray;
             }
 
-            var sortedJArray = new JArray();
-            foreach (JToken item in jArray)
+            using (var stream = new MemoryStream())
             {
-                if (item is JObject elem)
-                {
-                    sortedJArray.Add(elem.SortProperties());
-                }
-                else if (item is JArray arrayElem)
+                using (var writer = new Utf8JsonWriter(stream))
                 {
-                    sortedJArray.Add(arrayElem.SortProperties());
+                    writer.WriteStartArray();
+                    foreach (JsonElement item in jArray.EnumerateArray())
+                    {                     
+                        if (item.ValueKind == JsonValueKind.Object)
+                        {
+                            item.SortProperties().WriteTo(writer);
+                        }
+                        else if (item.ValueKind == JsonValueKind.Array)
```

Comment:
What happens if a JSON array contains primitives like a list of strings? Looks like sort is ignoring them. Is that intentional?

```C#
string json = "{\"a\":1, \"c\": [1, 3, 5], \"b\": 3}";
using var doc = JsonDocument.Parse(json);
Console.WriteLine(doc.RootElement.SortProperties().ToString()); // {"a":1,"b":3,"c":[]}
```